### ISSUE TITLE
Custom GraphNode not displaying slots

### ISSUE THREAD

**Issue by dploeger** [godotengine/godot#12633]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**

* macOS 10.13, Godot 3.0.alpha.build2

**Issue description:**

I'm currently writing a custom editor plugin with which it should be easy to create maps for discovery puzzle games (i.e. Myst-like).

I thought, that the GraphEditor/GraphNode objects should be a good way of implementing that. However, my custom GraphNode is not displayed and handled correctly in the GraphEditor. It's just a black rectangle without any slots to manage ~~or title displayed~~ (handled in #12637):

![image](https://user-images.githubusercontent.com/203453/32406157-7ec3cb6a-c173-11e7-9845-02c451f50de2.png)

I tried to work closely referencing [StraToN's Dialogtree plugin](https://github.com/StraToN/godot-dialogtreetools), which at least works in Godot 2 (it's not ported to 3 yet).

Maybe I'm missing some implementation step, but the documentation is not telling me more currently. (I will update it, though, if you'll point me to the correct way of implementing that)

**Steps to reproduce:**

* Unzip the attached Compass.zip and open the project in Godot 3
* Enable the included plugin
* Open the Map Editor panel in the lower panel
* Click on "Add Node"

**Link to minimal example project:**
[Compass.zip](https://github.com/godotengine/godot/files/1443250/Compass.zip)

**Comment by rodolforg** [comment]
I was reporting an issue that can be your case too.
Try this: node->add_color_override("title_color", Color(1, 1, 1));

**Comment by djrm** [comment]
yes, there is node "title_color" and "close color" (for the close button), i think they should be changed so that default shows something.

**Comment by dploeger** [comment]
Yes. That shows the title and the close button, but the close button is not working (which could be expected as the close_request signal has to be catched and implemented)

However, there still are no slots available or displayed.

**Comment by rodolforg** [comment]
@dploeger maybe it is also related to default colors.
Try something like this (in C++)
`node->set_slot(0, false, 0, Color(), true, 1000, Color(1,1,1));`

**Comment by dploeger** [comment]
@rodolforg I already have this in my code:
```
	waypoint.set_slot(0, true, 0, Color(1,0,0), true, 0, Color(0,1,0))
	waypoint.set_slot(1, true, 0, Color(0,0,1), true, 0, Color(1,1,0))
```
Doesn't work.

**Comment by rodolforg** [comment]
You also need to add a label o.O
```
var text = Label.new()
text.set_text("label")
text.set_align("right")
waypoint.add_child(text)
```

**Comment by dploeger** [comment]
Argh. Yes, that works. Hm. I guess, this is part documentation, part bug, right? Something around like "To name your slots, add labels appropiately", but even without labels, the slots should be displayed, right?

**Comment by rodolforg** [comment]
I suppose they should, but...

**Comment by Jimmio92** [comment]
This is still a problem, and I for sure see this as a bug.

1) You cannot add more than one slot unless there's an item there (label works great, but...)
2) You cannot add other elements where the labels are, so there's a huge gap in the GraphNode
3) If there are inputs and outputs, labels have to be created that overlap or the text for the left has to be in the right label and vice versa.

I think the labels should be a **part of** the slot system, and I think they should exist on a separate layer so as to not interfere with other UI elements added to the container. The container should auto-resize to fit your content minimum size, plus the labels on each side and their slot connectors. Possibly, the GraphNode should not override margin and anchor information, allowing UI elements to be placed however is best for the user.

**Comment by KoBeWi** [comment]
The documentation part is resolved by #34372

If you think that GraphNode needs improvements, create a new issue at https://github.com/godotengine/godot-proposals/issues
